pkg/workflow/controllers/chaos_node_reconciler.go

```diff
@@ -53,20 +62,122 @@ func (it *ChaosNodeReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	if ConditionEqualsTo(node.Status, v1alpha1.ConditionDeadlineExceed, corev1.ConditionTrue) {
-		err := it.recoverChaos(ctx, node)
+	it.logger.V(4).Info("resolve chaos node", "node", request)
+
+	err = it.syncChaosResources(ctx, node)
+	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	if !ConditionEqualsTo(node.Status, v1alpha1.ConditionChaosInjected, corev1.ConditionTrue) {
-		err = it.injectChaos(ctx, node)
-		return reconcile.Result{}, err
+	updateError := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		nodeNeedUpdate := v1alpha1.WorkflowNode{}
+		err := it.kubeClient.Get(ctx, request.NamespacedName, &nodeNeedUpdate)
+		if err != nil {
+			return client.IgnoreNotFound(err)
+		}
+
+		chaosList, err := it.fetchChildrenChaosCustomResource(ctx, nodeNeedUpdate)
+		if err != nil {
+			return client.IgnoreNotFound(err)
+		}
+		if len(chaosList) > 1 {
```

Comment:
Why do we not need this log when `len(chaosList) == 1` or `len(chaosList) == 0`?

Reply:
because `len(chaosList) > 1` is not expected, and only happens when the cache of controller-runtime is oudated. It's kind of a WARN log.